code/game/dna/genes/vg_powers.dm

```diff
@@ -158,3 +158,133 @@ var/noir_master = list(new /obj/abstract/screen/plane_master/noir_master(),new /
+		return // no brain means no explosion
+
+	var/datum/organ/external/head/head_organ = M.get_organ(LIMB_HEAD)
+	var/obj/item/organ/internal/brain/B = M.remove_internal_organ(M, M.internal_organs_by_name["brain"], head_organ)
+	M.death(0)
+	M.ghostize(0) // set so they can't come back
+	head_organ.explode()
+	fire(M)
+	qdel(B) // make sure the actual brain doesn't drop
+	log_admin("[key_name(M)] has launched their brain! ([formatJumpTo(M)])")
+	message_admins("[key_name(M)] has launched their brain! ([formatJumpTo(M)])")
+
+/spell/targeted/genetic/headcannon/proc/fire(var/mob/living/carbon/human/user)
+	// stolen/adapted from the wheelchair cannon
+	var/obj/item/projectile/brain/cannonbrain = new()
+	var/target = null
+	switch(user.dir)
+		if(1)
+			target = locate(user.x, user.y+20, user.z)
+		if(2)
+			target = locate(user.x, user.y-20, user.z)
+		if(4)
+			target = locate(user.x+20, user.y, user.z)
+		if(8)
+			target = locate(user.x-20, user.y, user.z)
+
+	var/turf/curloc = get_turf(user)
+	var/turf/targloc = get_turf(target)
+	if (!istype(targloc) || !istype(curloc))
+		return
+
+	user.visible_message("<span class='danger'>Something explodes out of [user]'s forehead!</span>","<span class='danger'>You fire your brain from your head!</span>")
+	log_attack("[user.name] ([user.ckey]) fired their brain (proj:[cannonbrain.name]) at coordinates ([user.x],[user.y],[user.z])" )
+
+	// recoil body (copied from code/module/projectile/gun.dm)
+	var/movementdirection = get_dir(target,user)
+	spawn()
+		shake_camera(user, 6, 5)
+	if(user.locked_to && isobj(user.locked_to) && !user.locked_to.anchored)
+		spawn()
+			var/obj/B = user.locked_to
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(1)
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(1)
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(1)
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(2)
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(2)
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(3)
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(3)
+			B.Move(get_step(user,movementdirection), movementdirection)
+			sleep(3)
+			B.Move(get_step(user,movementdirection), movementdirection)
+	else
+		user.Move(get_step(user,movementdirection), movementdirection)
+		sleep(1)
+		user.Move(get_step(user,movementdirection), movementdirection)
+	user.apply_inertia(movementdirection)
+
+	// projectile stuff
+	cannonbrain.def_zone = LIMB_CHEST // always target chest
+	cannonbrain.original = target
+	cannonbrain.forceMove(curloc)
+	cannonbrain.starting = curloc
+	cannonbrain.shot_from = user
+	cannonbrain.firer = user
+	cannonbrain.current = curloc
+	cannonbrain.OnFired()
+	cannonbrain.yo = targloc.y - curloc.y
+	cannonbrain.xo = targloc.x - curloc.x
+	playsound(user, cannonbrain.fire_sound, 50, 1)
+	spawn()
+		cannonbrain.process()
```

Comment:
I was largely going off of the `Fire` proc in gun.dm for this part. I don't believe that the projectile is otherwise processed.

Reply:
Gun code is horrible
This would call the process() of the thing exactly only once
You may want `processing_objects += cannonbrain`

If you're going this way, it doesn't really need to be behind a `spawn()` since it'll process on next tick.

Reply:
Ignore sh*fty, this is how projectiles work. This should be fine.